### ISSUE TITLE
B018 wrongly detects inline variable or attribute docstrings

### ISSUE THREAD

**Issue by xqt** [PyCQA/flake8-bugbear#208]
Having an inline attribute doc string or a module variable docstring, it is wrongly marked as B018 (sample from sphinx doc):

```
module_level_variable2 = 98765
"""int: Module level variable documented inline.

The docstring may span multiple lines. The type may optionally be specified
on the first line, separated by a colon.
"""

```

**Comment by PhilippSelenium** [comment]
Can confirm.

**Comment by Mischback** [comment]
Can confirm, too.

You may see a failing Github Action run here: https://github.com/Mischback/django-calingen/actions/runs/1514981629

```
# FLAKE 8 SETTINGS regarding plugin configuration

# Activate rules by plugins
extend-select =
    # flake8-assertive
    A,
    # flake8-bugbear
    B, B902, B903,
    # pydocstyle (by flake8-docstrings)
    D,
    # flake8-django
    DJ, DJ10, DJ11,
```

The full ``flake8 configuration`` can be found here: https://github.com/Mischback/django-calingen/blob/interfaces/.flake8

As ``flake8-bugbear`` is run by ``flake8`` pre-commit hook, this is the relevant part of my ``pre-commit configuration``:

```
repos:
  # ... cut ...
  # flake8 (linting)
  - repo: https://github.com/PyCQA/flake8
    rev: 4.0.1
    hooks:
      - id: flake8
        additional_dependencies: [flake8-bugbear, flake8-docstrings, flake8-django, flake8-assertive]
```

The full ``pre-commit configuration`` can be found here: https://github.com/Mischback/django-calingen/blob/interfaces/.pre-commit-config.yaml

**Comment by henzef** [comment]
B018 is also triggered for multi-line strings that are used as a comment aside from docstrings. Is this behaviour intentional?

**Comment by oliver-sanders** [comment]
For info, attribute docstrings [aren't a Python feature](https://www.python.org/dev/peps/pep-0224/), however, are widely used and supported by the Sphinx documentation system via [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoproperty).

**Comment by cooperlees** [comment]
Damn. Sorry all, definitely missed this and it's not intended. A PR adding a unit test showing we no longer flag this will definitely be accepted. I'll try get to it but welcome to beat me to it!

cc: @kasium - Checks creator to see if they can workout a fix quicker than I :D

**Comment by kasium** [comment]
@cooperlees I'm very sorry for that. As a fast workaround, let's just skip strings and work with anything else. Then we can look for a better solution

**Comment by kasium** [comment]
Hi all. I created #209

**Comment by cooperlees** [comment]
21.11.29 released reverting checking of strings for now. We will work on adding some pattern matching support into the string detection and probably default to ignoring this edge use case.

I also, 21.11.28 yanked from PyPI so it can not be installed to create noise.

**Comment by Mischback** [comment]
Wow, that was fast!

@cooperlees @kasium Thx a lot!

**Comment by cooperlees** [comment]
CI unwanted noise is the worst so I didn’t want to create it. This needs more thought and this will give us time rather than rushing out something.

**Comment by kasium** [comment]
@cooperlees the issue is, that the used quotes (""" or " or ') are not accessible in the AST. Therefore we need to use tokens. Since it's used for docstrings even in the middle of code, I would suggest to keep strings out. If it's fine for you, I would enhance the tests, code and README to reflect it.

**Comment by cooperlees** [comment]
So you think there's not way to look at the string contents themselves by keywords or regex to try reduce the noise and reenable string? I was thinking of (worst case) a way for people to specify from regex's and have some defaults ...

**Comment by kasium** [comment]
I think that this is somehow error prone. What's possible is, to only allow this pattern in certain cases, e.g. on module level and in \_\_init__ methods. So, e.g. in the middle of a custom function such a string makes no sense. 

However, some users use multiline strings as comments...

**Comment by Mischback** [comment]
> I think that this is somehow error prone. What's possible is, to only allow this pattern in certain cases, e.g. on module level and in __init__ methods. So, e.g. in the middle of a custom function such a string makes no sense.

Personally, I use ``numpy doc style`` documentation in combination with ``Sphinx``, ``sphinx-autodoc`` on a ``Django`` project. The attribute docstrings are used for class variables (or, in particular: ``Django model fields``).

> However, some users use multiline strings as comments...

I think this is _bad behaviour_ and can **and should** be marked by the corresponding rule, perhaps even with the _hint_ to use ``# ...`` lines for that.

**Comment by johnthagen** [comment]
I am hitting this issue for format strings used as attribute docstrings.

Note that attribute docstrings are listed as an official convention in [PEP 257](https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring).

> String literals occurring immediately **after a simple assignment at the top level** of a module, **class**, or __init__ method are called "attribute docstrings".

Here is a false positive:

```py
from dataclasses import dataclass

x = 0


@dataclass
class Foo:
    bar: int
    f"""Attribute docstring {x}"""
```

Output:

```
main.py:9:5: B018 Found useless expression. Either assign it to a variable or remove it.
```

@cooperlees Do *format *string checks need to be temporarily rolled back too?

### Environment

- Python 3.9.8

```
pip list
Package        Version
-------------- --------
attrs          21.4.0
click          8.0.1
flake8         4.0.1
flake8-bugbear 21.11.29
mccabe         0.6.1
pep517         0.10.0
pip            21.3.1
pip-tools      6.4.0
pycodestyle    2.8.0
pyflakes       2.4.0
setuptools     57.0.0
toml           0.10.2
wheel          0.36.2
```

**Comment by kasium** [comment]
My fault. The ignore was only for strings and not for f-strings (aka JoinedStr). PR is open
